sssd_test_framework/hosts/ldap.py

```diff
@@ -35,6 +42,11 @@ def __init__(self, *args, **kwargs) -> None:
         # Backup of original data
         self.__backup: dict[str, dict[str, list[bytes]]] = {}
 
+    def setup(self) -> None:
+        # Make sure ldap is running for each test
+        self._start_and_wait()
+        super().setup()
```

Comment:
Switch the order please. Make sure to run parent setup as the first step (and parent teardown as the last step for teardowns).

Reply:
Actually, you can just remove this method from LDAPHost. You already call _start_and_wait in backup() and the call stack is:

```
BaseBackupHost.setup() -> LDAPHost.backup() -> LDAPHost._start_and_wait()
```

Reply:
That will not work, backup is executed only once so only first test will start ldap and for other ones there will be 
no guarantee that it still runs.

I can switch the order through as the order will be:
LDAPHost.setup()->BaseBackupHost.setup()->LDAPHost.backup() {running the LDAPHost._start_and_wait() itself}

Reply:
Reordered.